src/gsy_e/models/strategy/__init__.py

```diff
@@ -752,6 +752,7 @@ def post_bid(
         Returns: The bid posted to the market
 
         """
+        self._assert_market_type_on_bid_event(market.id)
```

Comment:
The name of the method states that this should be executed only after a bid event, in fact before this PR it was only called in `event_bid_*` methods.
I'd recommend changing the name (e.g. something like `validate_bid_market_type` or `bid_can_happen_in_market`, or anything else).
Also, I think this method is a good candidate to become a decorator. What do you think?

Reply:
I agree on the renaming and would choose `validate_bid_can_be_posted_on_market`. The decorator solution does not work because the interfaces of the functions that this method is used in is different (sometime the BaseMarket object is pass, sometime only the market id).

Reply:
I agree with the general renaming idea, however I would recommend to stick to assert in the name to denote that an AssertionError is expected as the only "output" of the method. `_assert_bid_can_be_posted_on_market` for instance would work for me

Reply:
That's exactly the reason why it should be an assertion, that this should never be triggered by end-user settings.